bin/main.rb

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/ruby
+#
+require_relative '../lib/tictactoe'
+
+class MainIO
+  def initialize(stdin: '', stdout: [])
+    @stdin = stdin
+    @stdout = stdout
+    @accepted_input = [*1..9].map!(&:to_s)
+  end
+
+  def output_to_user(string)
+    @stdout.puts string
+  end
+
+  def prompt_user_for_position
+    input = $stdin.gets.chomp
+    int_to_return = @accepted_input.include?(input) ? input.to_i - 1 : nil
```

Comment:
The fact that this line subtracts 1 from the input number seems like a potential mash-up of interface-focused behavior vs knowledge specific to the TTT implementation (namely that you're using the input as an array index). A cleaner separation might have this class just validating the input and returning it unchanged.